docs/content/guides/developer/app-examples/trustless-swap.mdx

```diff
@@ -1048,741 +1275,1284 @@ Note that `dApp Kit` provides a set of hooks for making query and mutation calls
 
 :::
 
-### Connect wallet
+<details>
+  <summary>
+    Create `src/components/Header.tsx`. This file contains the navigation links and the connect wallet button:
+  </summary>
+```tsx
+import { ConnectButton } from "@mysten/dapp-kit";
+import { SizeIcon } from "@radix-ui/react-icons";
+import { Box, Container, Flex, Heading } from "@radix-ui/themes";
+import { NavLink } from "react-router-dom";
+
+const menu = [
+  {
+    title: "Escrows",
+    link: "/escrows",
+  },
+  {
+    title: "Manage Objects",
+    link: "/locked",
+  },
+];
+
+export function Header() {
+  return (
+    <Container>
+      <Flex
+        position="sticky"
+        px="4"
+        py="2"
+        justify="between"
+        className="border-b flex flex-wrap"
+      >
+        <Box>
+          <Heading className="flex items-center gap-3">
+            <SizeIcon width={24} height={24} />
+            Trading Demo
+          </Heading>
+        </Box>
+
+        <Box className="flex gap-5 items-center">
+          {menu.map((item) => (
+            <NavLink
+              key={item.link}
+              to={item.link}
+              className={({ isActive, isPending }) =>
+                `cursor-pointer flex items-center gap-2 ${
+                  isPending
+                    ? "pending"
+                    : isActive
+                      ? "font-bold text-blue-600"
+                      : ""
+                }`
+              }
+            >
+              {item.title}
+            </NavLink>
+          ))}
+        </Box>
+
+        <Box className="connect-wallet-wrapper">
+          <ConnectButton />
+        </Box>
+      </Flex>
+    </Container>
+  );
+}
+```
+</details>
 
 The dApp Kit comes with a pre-built React.js component called `ConnectButton` displaying a button to connect and disconnect a wallet. The connecting and disconnecting wallet logic is handled seamlessly so you don't need to worry about repeating yourself doing the same logic all over again.
 
-Place the `ConnectButton` in the header:
+:::danger[Checkpoint]
```

Comment:
Love the idea of having a checkpoint but we may need to rethink the icon 😅

Reply:
@ronny-mysten